### PR TITLE
remove usage of jQuery 'hover'

### LINKED ISSUE #1369: Warnings from jQuery Migrate helper
### Reproduction Steps

Install and activate the plugin: [Enable jQuery Migrate Helper](https://wordpress.org/plugins/enable-jquery-migrate-helper/)
Go to LifterLMS >> Settings and see the Warning as shown in the screenshot:
image

### Expected Behavior

Remove deprecated code

### Actual Behavior

Still has deprecated code

### Error Messages / Logs

+ Include any relevant error messages or log files
```
N/A

```

### System and Environment Information

<details>
<summary>System Report</summary>

```
System Report:
Wordpress
-------------------------------------------

Home Url: [removed]
Site Url: [removed]
Login Url: [removed]/wp-login.php
Version: 5.5.1
Debug Mode: No
Debug Log: No
Debug Display: Yes
Locale: en_CA
Multisite: No
Page For Posts: Not Set
Page On Front: Home (#5) [[removed]/]
Permalink Structure: /%postname%/
Show On Front: page
Wp Cron: Yes


Settings
-------------------------------------------

Version: 4.5.0
Db Version: 4.5.0
Course Catalog: Online Courses (#328) [[removed]/course-information/courses/]
Membership Catalog: Memberships (#3046) [[removed]/course-information/membership-catalog/]
Student Dashboard: My Dashboard (#331) [[removed]/dashboard/]
Checkout Page: Checkout Purchases (#330) [[removed]/dashboard/checkout/]
Course Catalog Per Page: 2
Course Catalog Sorting: menu_order,ASC
Membership Catalog Per Page: 9
Membership Catalog Sorting: menu_order,ASC
Site Membership: Not Set
Courses Endpoint: my-courses
Edit Endpoint: edit-account
Lost Password Endpoint: lost-password
Vouchers Endpoint: redeem-voucher
Autogenerate Username: yes
Password Strength Meter: yes
Minimum Password Strength: strong
Terms Required: yes
Terms Page: Terms and Conditions (#663) [[removed]/dashboard/terms-and-conditions/]
Checkout Names: required
Checkout Address: required
Checkout Phone: required
Checkout Email Confirmation: yes
Open Registration: yes
Registration Names: required
Registration Address: required
Registration Phone: required
Registration Voucher: optional
Registration Email Confirmation: yes
Account Names: required
Account Address: required
Account Phone: required
Account Email Confirmation: yes
Confirmation Endpoint: confirm-payment
Force Ssl Checkout: yes
Country: CA
Currency: CAD
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Decimals: 2
Trim Zero Decimals: no
Recurring Payments: yes
Email From Address: [removed]
Email From Name: [removed]
Email Footer Text: [removed]
Email Header Image: 425
Cert Bg Width: 720
Cert Bg Height: 577
Cert Legacy Compat: yes


Gateways
-------------------------------------------

PayPal: Enabled
PayPal Test Mode: Disabled
PayPal Logging: no
PayPal Order: 1
Manual: Enabled
Manual Logging: no
Manual Order: 1


Server
-------------------------------------------

Mysql Version: 5.5.5
Php Curl: Yes
Php Default Timezone: UTC
Php Fsockopen: Yes
Php Max Input Vars: 3000
Php Max Upload Size: 24 MB
Php Memory Limit: 256M
Php Post Max Size: 24M
Php Soap: No
Php Suhosin: No
Php Time Limt: 150
Php Version: 7.4.11
Software: Apache
Wp Memory Limit: 40M


Browser
-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0


Theme
-------------------------------------------

Name: Astra
Version: 2.5.5
Themeuri: https://wpastra.com/
Authoruri: https://wpastra.com/about/
Template:
Child Theme: No
Llms Support: Yes


Plugins
-------------------------------------------

Accessibility by UserWay: 1.2
Advanced Database Cleaner: 3.0.3
Akismet Anti-Spam: 4.1.6
All In One WP Security: 4.4.4
Anti-Malware Security and Brute-Force Firewall: 4.19.69
Better Search Replace: 1.3.3
Broken Link Checker: 1.11.14
Contact Form 7: 5.2.2
Cookie Notice: 1.3.2
Definitely allow mobile zooming: 1.4
Display Environment Type: 1.3
Duplicate TEC Event: 1.5.2
Email Log: 2.4.3
Enable jQuery Migrate Helper: 1.0.1
Events Shortcodes and Templates Addon: 1.5.3
Event Tickets: 5.0.1
Event Tickets/Eventbrite Extension: Show Cost Field: 1.0.1
GTranslate: 2.8.60
HTML5 Responsive FAQ: 2.8.5
LifterLMS: 4.5.0
LifterLMS Advanced Quizzes: 1.1.0
LifterLMS Helper: 3.1.0
LifterLMS Labs: 1.5.3
LifterLMS PayPal Gateway: 1.3.0
Login-Logout: 3.8
Log Out Shortcode: 1.0.8
Meta Tag Manager: 2.1.2
Rank Math SEO: 1.0.48.2
Shortcode For Current Date: 2.1.2
Shortcoder: 5.3.2
Simple Social Icons: 3.0.2
The Events Calendar: 5.2.0
The Events Calendar Extension: Events Control: 1.0.0
The Events Calendar Shortcode & Block: 2.4
Ultimate Addons for Gutenberg: 1.17.0
UpdraftPlus - Backup/Restore: 1.16.29
WP Content Copy Protection with Color Design: 2.2.7
WP GDPR Compliance: 1.5.4
WP Super Cache: 1.7.1


Integrations
-------------------------------------------

BbPress: No
BuddyPress: No


Template Overrides
-------------------------------------------


```

</details>

This issue has be recreated:
+ [x] Locally
+ [x] On a staging site
+ [x] On a production website
+ [ ] With only LifterLMS and a default theme

### Browser, Device, and Operating System Information

+ Browser name and version
+ Operating System name and version
+ Device name and version (if applicable)
HS-137303

### PR DESCRIPTION
**Pull request by thomasplevy**
## Description

Fixes #1369 

## How has this been tested?

+ Manually

## Types of changes
+ Remove usage of deprecated jQuery event

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: assets/js/llms-admin.js [hunk 1]
```diff
@@ -13,39 +13,46 @@
...
+		};
 
-				}
+		this.bind = function() {
 
+			this.$info_toggles.on( 'mouseenter', function() {
+				toggleWidgetInfo( $( this ), 'show' );
+			} ).on( 'mouseleave', function() {
+				toggleWidgetInfo( $( this ), 'hide' );
 			} );
```
Comment (eri-trabiccolo) [inline assets/js/llms-admin.js]:
```suggestion
			this.$info_toggles.on( 'mouseenter mouseleave', function( evt ) {
				$(this).closest( '.llms-widget' )
					.toggleClass( 'info-showing', 'mouseenter' === evt.type );
			} );
```

File: assets/js/llms-admin.js [hunk 2]
```diff
@@ -13,39 +13,46 @@
...
+		/**
+		 * Toggle display of info widget description tips
+		 *
+		 * @since [version]
+		 *
+		 * @param {Object} $toggle The jQuery selector for the toggle element.
+		 * @param {String} action  Requested action, either "hide" or "show".
+		 * @return {void}
+		 */
+		function toggleWidgetInfo( $toggle, action ) {
```
Comment (eri-trabiccolo) [inline assets/js/llms-admin.js]:
```suggestion
```

File: assets/js/llms-admin.js [hunk 3]
```diff
@@ -13,39 +13,46 @@
...
 
-				var $toggle = $( this ),
-					$widget = $toggle.closest( '.llms-widget' ),
-					$info = $widget.find( '.llms-widget-info' ),
-					action = ( $widget.hasClass( 'info-showing' ) ) ? 'hide' : 'show';
+			var $widget = $toggle.closest( '.llms-widget' ),
+				$info   = $widget.find( '.llms-widget-info' );
```
Comment (eri-trabiccolo) [inline assets/js/llms-admin.js]:
```suggestion
```

File: assets/js/llms-admin.js [hunk 4]
```diff
@@ -13,39 +13,46 @@
...
 
-				self.$widgets.removeClass( 'info-showing' );
+			if ( 'show' === action ) {
+				$widget.addClass( 'info-showing' );
+			} else if ( 'hide' === action ) {
+				$widget.removeClass( 'info-showing' );
+			}
```
Comment (eri-trabiccolo) [inline assets/js/llms-admin.js]:
```suggestion
```

File: assets/js/llms-admin.js [hunk 5]
```diff
@@ -13,39 +13,46 @@
...
 
-				if ( 'show' === action ) {
+		};
```
Comment (eri-trabiccolo) [inline assets/js/llms-admin.js]:
```suggestion
```

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
I proposed some golf changes, but maybe they're hard to read...
Basically my suggestion is to make the whole `window.llms.widget` looking like this:

```js
	window.llms.widgets = function() {

		this.$widgets      = $( '.llms-widget' );
		this.$info_toggles = $( '.llms-widget-info-toggle' );

		this.init = function() {
			this.bind();
		};

		this.bind = function() {

				this.$info_toggles.on( 'mouseenter mouseleave', function( evt ) {
					$(this).closest( '.llms-widget' )
						.toggleClass( 'info-showing', 'mouseenter' === evt.type );
				} );

		};

		// Go.
		this.init();

		return this;

	};
```

what do you think?

**Comment by thomasplevy** [comment]
@eri-trabiccolo next time just write the pr yourself jeez.

**Pull request merged**